spec/errors.md

```diff
@@ -241,49 +241,72 @@ or for private implementation use that is not supported by the current implement
+>
+> ```
+> .some {|horse|}
+> {{The message body}}
+> ```
+
+## Message Function Errors
 
-An **_<dfn>Invalid Expression</dfn>_** error occurs when a _message_ includes an _expression_
+### Bad Expression
+
+A **_<dfn>Bad Expression</dfn>_** error occurs when a _message_ includes an _expression_
 whose implementation-defined internal requirements produce an error during _function resolution_
 or when a _function_ returns a value (such as `null`) that the implementation does not support.
 
-An **_<dfn>Operand Mismatch Error</dfn>_** is an _Invalid Expression_ error that occurs when
+> For example, the following _message_ might produce a _Bad Expression_ error if the
+> the function `:function` threw an exception or otherwise emitted an error
+> rather than returning a valid value:
+>
+> ```
+> This has a bad expression {$var :function} because it has a bug in it.
+> ```
+
+### Bad Operand
+
+A **_<dfn>Bad Operand</dfn>_** error is an error that occurs when
 an _operand_ provided to a _function_ during _function resolution_ does not match one of the
 expected implementation-defined types for that function;
 or in which a literal _operand_ value does not have the required format
 and thus cannot be processed into one of the expected implementation-defined types
 for that specific _function_.
 
-> For example, the following _message_ produces an _Operand Mismatch Error_
-> (a type of _Invalid Expression_ error)
+> For example, the following _message_ produces a _Bad Operand_ error
 > because the literal `|horse|` does not match the production `number-literal`,
 > which is a requirement of the function `:number` for its operand:
+>
 > ```
 > .local $horse = {horse :number}
 > {{You have a {$horse}.}}
 > ```
-> The following _message_ might produce an _Invalid Expression_ error if the
-> the function `:function` threw an exception or otherwise emitted an error
-> rather than returning a valid value:
->```
-> {{This has an invalid expression {$var :function} because it has a bug in it.}}
->```
 
-### Unsupported Statement
+### Bad Option
 
-An **_<dfn>Unsupported Statement</dfn>_** error occurs when a message includes a _reserved statement_.
+A **_<dfn>Bad Option</dfn>_** error is an error that occurs when
+an _option_ value provided to a _function_ during _function resolution_ does not match one of the
+expected implementation-defined types for that function;
+or in which a literal _option_ value does not have the required format
+and thus cannot be processed into one of the expected implementation-defined types
+for that specific _function_.
 
-> For example, attempting to format this message
-> would always result in an _Unsupported Statement_ error:
+> For example, the following _message_ might produce a _Bad Option_ error
+> because the literal `foo` does not match the production `number-literal`,
+> which is a requirement of the function `:number` for its `minimumFractionDigits` _option_:
```

Comment:
Slightly wrong. `|foo|` doesn't match [digit size option](https://github.com/unicode-org/message-format-wg/blob/main/spec/registry.md#digit-size-options)

Reply:
```suggestion
> because the literal `foo` does not match the production `digit-size-option`,
> which is a requirement of the function `:number` for its `minimumFractionDigits` _option_:
```